app/src/main/java/uw/hcrlab/kubi/lesson/prompts/TranslatePrompt.java

```diff
@@ -76,17 +67,58 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         return view;
     }
 
-    private String generateButtonTag(String text, int idx) {
-        return text + "-" + idx;
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        robot.loadPronunciation(PromptData.combineWords(this.data.words));
     }
 
     @Override
-    public void onClick(View v) {
-        Log.i(TAG, "clicked " + wordsById.get(v.getId()));
-        TextView button = (TextView) v;
-        PromptData.Word word = wordsById.get(v.getId());
-        // can safely assume the word has hints; it if didn't it would not be clickable
-        robot.showHint(word.hints);
+    public void onResume() {
+        super.onResume();
+
+        robot.pronounceAfterSpeech(PromptData.combineWords(this.data.words));
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+
+        for(PromptData.Word word : this.data.words) {
+            if(word.hasHint()) {
+                robot.unloadPronunciation(word.text);
+            }
+        }
+    }
+
+    private View getWordView(LayoutInflater inflater, ViewGroup container, PromptData.Word word) {
+        View view;
+
+        if(word.hasHint()) {
+            Log.i(TAG, "Creating word button: " + word.text);
+
+            int id = View.generateViewId();
+            wordsById.put(id, word);
+
+            robot.loadPronunciation(word.text);
+
+            view = inflater.inflate(R.layout.word_button, container, false);
+
+            Button wordButton = (Button) view.findViewById(R.id.button);
+            wordButton.setText(word.text);
+            wordButton.setId(id);  // TODO: is this safe? (http://tinyurl.com/lzeu2at suggests it is)
```

Comment:
I meant to ask your opinion on this line (that I wrote) earlier... I'm using the ID to figure out which button was clicked in the hintClickListener. It seems sort of iffy to set a layout element's ID when the layout element has a hard-coded ID in its .xml file. But it was the only way I could figure out to identify which button was clicked. 

Any thoughts?